### ISSUE TITLE
Consistency of NArray bracket accesses

### ISSUE THREAD

**Issue by fmoessbauer** [dash-project/dash#324]
During the implementation of the dash coarray for atomic types, I noticed that the following accesses return different types:

```cpp
dash::NArray<int,2> matrix(5,10);
auto a = matrix(0,0);  // returns a GlobRef<int>
auto b = matrix[0][0]; // returns a MatrixRef<int, 2, 0, PatternT>
```

Is this really desired? I would appreciate it if both accesses return a GlobRef, as the GlobRef has specializations for atomic accesses. Otherwise we have to implement the atomic specializations also for the `MatrixRef<int,0,...>`

**Comment by devreal** [comment]
Afaics, `MatrixRef::operator[]()` returns again a `MatrixRef` to keep the `operator[]` chain going:
```
  MatrixRef<ElementT, NumDimensions, NumViewDim-1, PatternT>
    operator[](index_type n);
```

I guess we could provide an overload that returns a `GlobRef` if `(NumViewDim-1) == 0)`?

**Comment by fmoessbauer** [comment]
I am currently working on exactly this functionality. Seems to be not too hard, but anyway I am interested if this was a design decision or just a "bug".

For my Coarray implementation this functionality is essential, as I have to proxy all views otherwise...

**Comment by fuchsto** [comment]
Fixed in #326
